### PR TITLE
Further refinements to phone number parsing.

### PR DESCRIPTION
**Pull request by charlesmchen**
PTAL @michaelkirk

### CODE REVIEW

File: src/Contacts/PhoneNumber.m [hunk 1]
```diff
@@ -144,13 +144,17 @@ + (PhoneNumber *)tryParsePhoneNumberFromUserSpecifiedText:(NSString *)text {
         }
     };
 
-    if ([sanitizedString hasPrefix:@"+"]) {
-        // If the text starts with "+", just parse it.
-        tryParsingWithCountryCode(sanitizedString, [self defaultRegionCode]);
+    tryParsingWithCountryCode(sanitizedString, [self defaultRegionCode]);
```
Comment (charlesmchen) [inline src/Contacts/PhoneNumber.m:147]: I decided the ordering was wrong; this result should be first, not last.

File: src/Contacts/PhoneNumber.m [hunk 2]
```diff
@@ -144,13 +144,17 @@ + (PhoneNumber *)tryParsePhoneNumberFromUserSpecifiedText:(NSString *)text {
...
 
+    if ([sanitizedString hasPrefix:@"+"]) {
+        // If the text starts with "+", don't try prepending
+        // anything else.
         return result;
     }
 
+    // Try just adding "+" and parsing it.
+    tryParsingWithCountryCode([NSString stringWithFormat:@"+%@", sanitizedString], [self defaultRegionCode]);
```
Comment (charlesmchen) [inline src/Contacts/PhoneNumber.m:156]: The key fix: We should always try parsing by adding "+" regardless of whether it starts with the calling code that corresponds to the device's default region.

**Comment by charlesmchen** [comment]
The repro is: 

* Use a device registered with a +1 (US) phone number.
* Enter a +44 (UK) E164 phone number ___but without___ a "+" plus sign that corresponds to a signal account. 

The phone number should show up as a signal user.

**Review by michaelkirk — APPROVED** [review]
LGTM

**Pull request merged**